app/controllers/services/service_instances_controller.rb

```diff
@@ -175,6 +175,33 @@ def permissions(guid)
       end
     end
 
+    get '/v2/service_instances/:guid/service_keys', :service_keys
```

Comment:
Hey, @xingzhou, please remove this method and use the `define_attributes` method listed on the top of this file: https://github.com/cloudfoundry/cloud_controller_ng/blob/master/app/controllers/services/service_instances_controller.rb#L10-L16 to be consistent with service bindings. `define_attributes` will implement the nested relationship.

Adding `to_many :service_keys` will implement this entire method. Some of the tests that verify the attributes will need to be updated. Also, the one_to_many relationship of service instances to service keys should be added to the [service instance model](https://github.com/cloudfoundry/cloud_controller_ng/blob/master/app/models/services/service_instance.rb#L21) for this to work.

Keep the existing tests that you have written for `/v2/service_instance/:guid/service_keys` endpoint.

Thanks,
Jeff and @rmasand, Services API Team